### PR TITLE
COSBETA-31: Bulk upload JS error handling

### PR DESCRIPTION
**Pull request by raamSoftwire**
Add JS to run form validation client-side in the case where the user selects too many files.

## Description
- Add data attribute to view
- Add javascript file to handle file upload error
- Import javascript file into `shared.js`
## Checklist:

- [x] Automated checks are passing locally.
### General testing
- [x] Test without javascript
- [ ] Test on small screen
### Accessibility testing
- [ ] Tested with one screen reader
- [x] Disable css - does content make sense and appear in a logical order?
- [ ] Passes automated checker (automated in build or manual)

### CODE REVIEW

**Comment by raamSoftwire** [comment]
I have based this PR on top of the other COSBETA-31 branch to keep things clearer (hopefully).

**Comment by raamSoftwire** [comment]
There is still a problem that we can't trigger the `govuk-error-summary` for "too many files selected" from the JS code. I'm still not sure about the best way to proceed here and it would be good to chat over the options.

File: cosmetics-web/app/views/responsible_persons/notification_files/new.html.slim [hunk 1]
```diff
@@ -5,7 +5,7 @@
 .govuk-grid-row
   .govuk-grid-column-two-thirds
     = form_with url: responsible_person_notification_files_path(@responsible_person), method: :post,
-            local: true do |form|
+            data: {max_no_of_files: NotificationFile.get_no_of_files_limit}, id: "file-upload-form", local: true do |form|
```
Comment (tom-bell-softwire) [inline cosmetics-web/app/views/responsible_persons/notification_files/new.html.slim]: I think we've used hypenated names for data attributes elsewhere. For consistency, something like `max-number-of-files` might be better.

File: cosmetics-web/app/assets/application/javascripts/bulk_file_upload_error_handling.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+$(document).ready(() => {
+    const fileUploadForm = document.getElementById('file-upload-form');
+    const fileInput = document.getElementById('uploaded_files');
+    const max_number_of_files = fileUploadForm.dataset.maxNoOfFiles;
+
+    fileUploadForm.addEventListener('submit', function (event) {
+        if (fileInput.files.length > max_number_of_files) {
+            event.preventDefault();
+            location.reload();
```
Comment (tom-bell-softwire) [inline cosmetics-web/app/assets/application/javascripts/bulk_file_upload_error_handling.js]: Does this show any feedback to the user (e.g. an error message or adding error styling to the file input field)?

**Comment by tom-bell-softwire** [comment]
Are all the items in the checklist above relevant for this ticket? Might be worth removing things about the changelog and some of the accessibility testing.

File: cosmetics-web/app/views/responsible_persons/notification_files/new.html.slim (hunk 1 above)
Comment (raamSoftwire) [reply cosmetics-web/app/views/responsible_persons/notification_files/new.html.slim]: No problem, I've changed that now.

File: cosmetics-web/app/assets/application/javascripts/bulk_file_upload_error_handling.js (hunk 2 above)
Comment (raamSoftwire) [reply cosmetics-web/app/assets/application/javascripts/bulk_file_upload_error_handling.js]: It does now

**Pull request merged**